packages/dapp/src/components/Main.tsx

```diff
@@ -1,21 +1,59 @@
 import * as React from "react";
 import { Switch, Route } from "react-router-dom";
 import Listings from "./Listings";
-import Newsroom from "./Newsroom";
+import Newsroom from "./newsroom/Newsroom";
 import Contracts from "./Contracts";
 import ContractPage from "./ContractPage";
+import Listing from "./listing/Listing";
+import Editor from "./Editor";
+import NewsroomManagement from "./newsroom/NewsroomManagement";
+import CreateNewsroom from "./CreateNewsroom";
+import Article from "./Article";
+import { setCivil } from "../helpers/civilInstance";
+
+export interface MainState {
+  civilSet: boolean;
+}
+class Main extends React.Component<{}, MainState> {
+  constructor(props: {}) {
+    super(props);
+    this.state = {
+      civilSet: false,
+    };
+  }
+  public componentWillMount(): void {
```

Comment:
componentDidMount.  react is removing all Will methods to support async rendering